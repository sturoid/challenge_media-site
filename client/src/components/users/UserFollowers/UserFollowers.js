import React from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import Loader from '../../../components/_shared/Loader';

export const UserFollowers = ({ followers }) => (
  <React.Fragment>
    <h2>Followers</h2>
    {followers.map(f => <div key={f.id}>{f.first_name}</div>)}
  </React.Fragment>
);

UserFollowers.propTypes = {
  followers: PropTypes.arrayOf(
    PropTypes.shape({
      first_name: PropTypes.string.isRequired
    })
  ).isRequired
};

class DataContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: false,
      error: false,
      followers: null
    };
  }

  componentDidMount() {
    this.fetchDataAndSetState();
  }

  async fetchDataAndSetState() {
    const { data: followers } = await axios.get(
      `/api/v1/user/${this.props.userId}/followers`
    );
    this.setState({ followers });
  }

  render() {
    const { loading, error, followers } = this.state;
    if (loading) return <Loader loading={loading} />;
    if (error) return <div>{error.message}</div>;
    return <UserFollowers followers={followers} />;
  }
}

DataContainer.propTypes = {
  userId: PropTypes.string.isRequired
};

export default DataContainer;
