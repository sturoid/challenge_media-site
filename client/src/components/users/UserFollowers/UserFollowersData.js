import React from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import { gql } from 'apollo-boost';
import Loader from '../../../components/_shared/Loader';
import UserFollowers from './UserFollowers';

const GET_USER_FOLLOWERS = gql`
  query userFollowers($id: String!) {
    userFollowers(id: $id) {
      id
      name
      location
      occupation
    }
  }
`;

const UserFollowersData = ({ userId }) => (
  <Query query={GET_USER_FOLLOWERS} variables={{ id: userId }}>
    {({ loading, error, data }) => {
      if (loading) return <Loader loading />;
      if (error) return `Error! ${error.message}`;
      return <UserFollowers followers={data.userFollowers} />;
    }}
  </Query>
);

UserFollowersData.propTypes = {
  userId: PropTypes.string.isRequired
};

export default UserFollowersData;
