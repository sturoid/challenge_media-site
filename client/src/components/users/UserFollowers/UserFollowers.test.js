import React from 'react';
import renderer from 'react-test-renderer';
import { UserFollowers } from './UserFollowers';
import followersData from '../../../../../lib/test-data/userFollowers';

describe('Rendering', () => {
  it('renders correctly and equals snapshot', () => {
    const tree = renderer
      .create(<UserFollowers followers={followersData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
