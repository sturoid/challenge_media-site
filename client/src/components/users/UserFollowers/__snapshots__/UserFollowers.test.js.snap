// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering renders correctly and equals snapshot 1`] = `
Array [
  <h2>
    Followers
</h2>,
  <div>
    Florian
</div>,
  <div>
    Amy
</div>,
  <div>
    Maxime
</div>,
  <div>
    Sean
</div>,
]
`;
